qmail-smtpd.c

```diff
@@ -217,14 +221,44 @@ int flagbarf; /* defined if seenmail */
 stralloc mailfrom = {0};
 stralloc rcptto = {0};
 
+#ifdef SMTPUTF8
+stralloc mfparms = {0};
+void mailfrom_parms(const char *arg)
+{
+  int i;
+  int len;
+
+  len = str_len(arg);
+  mfparms.len=0;
+  i = byte_chr(arg,len,'>');
+  if (i > 4 && i < len) {
+    while (len) {
+      arg++; len--;
+      if (*arg == ' ' || *arg == '\0' ) {
+         if (case_starts(mfparms.s,"SMTPUTF8")) smtputf8 = 1;
```

Comment:
I'm not sure if the case thing is actually right here. I have to reread the RfC, but I would bet this is not case insensitive.

Reply:
RFC 5321 2.4.  General Syntax Principles and Transaction Model

Verbs and argument values (e.g., "TO:" or "to:" in the RCPT command and extension name keywords) are not case sensitive, with the sole exception in this specification of a mailbox local-part.
SMTP Extensions may explicitly specify case-sensitive elements. But RFC 6531 doesn't mention any case requirement for SMTPUTF8 keyword. So I assume it inherits the definition from RFC 5321